scripts/eosio_build.sh

```diff
@@ -217,63 +227,80 @@ if [ $STALE_SUBMODS -gt 0 ]; then
+   while true; do
+   [[ $NONINTERACTIVE == 0 ]] && read -p "${COLOR_YELLOW}Do you wish to proceed anyway? (y/n)${COLOR_NC} " ANSWER
+   case $ANSWER in
+      "" ) echo "What would you like to do?";;
+      0 | true | [Yy]* ) break;;
+      1 | false | [Nn]* ) exit;;
+      * ) echo "Please type 'y' for yes or 'n' for no.";;
+   esac
+   done
 fi
 
-NO_CPP17=false
+# Checks for Arch and OS + Support for tests setting them manually
+## Necessary for linux exclusion while running bats tests/bash-bats/*.bash
+[[ -z "${ARCH}" ]] && export ARCH=$( uname )
+if [[ -z "${NAME}" ]]; then
+    if [[ $ARCH == "Linux" ]]; then
+        [[ ! -e /etc/os-release ]] && echo " - /etc/os-release not found! It seems you're attempting to use an unsupported Linux distribution." && exit 1
+        # Obtain OS NAME, and VERSION
+        . /etc/os-release
+    elif [[ $ARCH == "Darwin" ]]; then export NAME=$(sw_vers -productName)
+    else echo " - EOSIO is not supported for your Architecture!" && exit 1
+    fi
+fi
 
-WHICH_CPP=`which $CPP_COMP`
-COMPILER_TYPE=`readlink $WHICH_CPP`
-if [[ $COMPILER_TYPE == "clang++" ]]; then
-   if [[ `c++ --version | cut -d ' ' -f 1 | head -n 1` == "Apple" ]]; then
-      ### Apple clang version 10
-      if [[ `c++ --version | cut -d ' ' -f 4 | cut -d '.' -f 1 | head -n 1` -lt 10 ]]; then
-         NO_CPP17=true
+export BUILD_CLANG8=false
+export NO_CPP17=false
+
+export CXX=${CXX:-c++}
+export CC=${CC:-cc}
+if [[ $PIN_COMPILER == false ]]; then
+   which $CXX &>/dev/null || ( echo " - Unable to find compiler \"${CXX}\"! Pass in the -P option if you wish for us to install it OR set \$CXX to the proper binary."; exit 1 )
+   # readlink on mac differs from linux readlink (mac doesn't have -f)
+   [[ $ARCH == "Linux" ]] && READLINK_COMMAND="readlink -f" || READLINK_COMMAND="readlink"
+   COMPILER_TYPE=$( eval $READLINK_COMMAND $(which $CXX) )
+   [[ -z "${COMPILER_TYPE}" ]] && echo "COMPILER_TYPE not set!" && exit 1
+   if [[ $COMPILER_TYPE == "clang++" ]]; then
+      if [[ $ARCH == "Darwin" ]]; then
+            ### Check for apple clang version 10 or higher
+            [[ $( $(which $CXX) --version | cut -d ' ' -f 4 | cut -d '.' -f 1 | head -n 1 ) -lt 10 ]] && export NO_CPP17=true
+      else
+            ### Check for clang version 5 or higher
+            [[ $( $(which $CXX) --version | cut -d ' ' -f 4 | cut -d '.' -f 1 | head -n 1 ) -lt 5 ]] && export NO_CPP17=true
       fi
    else
-      ### clang version 5
-      if [[ `c++ --version | cut -d ' ' -f 4 | cut -d '.' -f 1 | head -n 1` -lt 5 ]]; then
-         NO_CPP17=true
-      fi
-   fi
-else
-   ### gcc version 7
-   if [[ `c++ -dumpversion | cut -d '.' -f 1` -lt 7 ]]; then
-      NO_CPP17=true
+      ## Check for c++ version 7 or higher
+      [[ $( $(which $CXX) -dumpversion | cut -d '.' -f 1 ) -lt 7 ]] && export NO_CPP17=true
    fi
+elif $PIN_COMPILER; then
+   export BUILD_CLANG8=true
+   export CPP_COMP=$CLANG8_ROOT/bin/clang++
```

Comment:
I think this can be:
```
CXX=$CLANG8_ROOT/bin/clang++
```

Reply:
That will cause cmake to not install (https://github.com/EOSIO/eos/issues/7260). That's why I set CXX later, after CLANG8 is built.